src/Kestrel.Core/HttpsConnectionAdapterOptions.cs

```diff
@@ -29,14 +30,25 @@ public HttpsConnectionAdapterOptions()
 
         /// <summary>
         /// <para>
-        /// Specifies the server certificate used to authenticate HTTPS connections.
+        /// Specifies the server certificate used to authenticate HTTPS connections. This is ignored if ServerCertificateSelector is set.
```

Comment:
I wish we didn't have to set an order of precedence. Are there any alternatives?

Reply:
There are several layers that can assign default certs, so the best option for now is to ignore it if a selector is provided. See https://github.com/aspnet/KestrelHttpServer/issues/2422